### PR TITLE
tidb-in-kubernetes: add tidb scheduler document

### PR DESCRIPTION
**Pull request by weekface**
### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [ ] This PR is to align with:
- [ ] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [x] dev (the latest development version)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

### CODE REVIEW

**Comment by cofyc** [comment]
shouldn't we link it in TOC?

**Comment by weekface** [comment]
> shouldn't we link it in TOC?

Done

File: dev/tidb-in-kubernetes/reference/components/tidb-scheduler.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+| 1  | 1  |
+| 2  | 1  |
+| 3  | 1  |
+| 4  | 1  |
+| 5  | 2  |
+| ...  |   |
+
+### TiKV 组件
+
+如果 K8s Node 节点数小于 3 个（K8s 集群 Node 数小于 3 个是无法实现 TiKV 高可用的），则可以任意调度；否则，每个 Node 节点上可调度的 TiKV 个数的计算公式为：Ceil (Replicas / 3) ，例如：
```
Comment (tennix) [inline dev/tidb-in-kubernetes/reference/components/tidb-scheduler.md]:
```suggestion
如果 K8s 节点数小于 3 个（K8s 集群节点数小于 3 个是无法实现 TiKV 高可用的），则可以任意调度；否则，每个节点上可调度的 TiKV 个数的计算公式为：Ceil (Replicas / 3) ，例如：
```

File: dev/tidb-in-kubernetes/reference/components/tidb-scheduler.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+| 4  | 2  | 1，1，2  |
+| 5  | 2  | 1，2，2  |
+| 6  | 2  | 2，2，2  |
+| 7  | 3  | 2，2，3  |
+| 8  | 3  | 2，3，3  |
+| ...  |   |   |
+
+### TiDB 组件
+
+实现了稳定调度：在 TiDB 组件滚动更新的时候，尽量将其调度回原来的 Node 节点，减少滚动更新时对集群的影响。
```
Comment (tennix) [inline dev/tidb-in-kubernetes/reference/components/tidb-scheduler.md]:
```suggestion
实现了稳定调度：在 TiDB 组件滚动更新的时候，尽量将其调度回原来的节点，减少滚动更新时对集群的影响。
```

Comment (tennix) [reply dev/tidb-in-kubernetes/reference/components/tidb-scheduler.md]: 可以简单说明一下这样做的原因，这个对于手动将 Node IP + NodePort 挂载在 LB 后端的场景比较有帮助，避免升级集群后 Node IP 发生变更需要重新调整 LB 配置

File: dev/tidb-in-kubernetes/reference/components/tidb-scheduler.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+## 工作原理
+
+![TiDB Scheduler Overview](/media/tidb-scheduler-overview.png)
+
+TiDB Scheduler 通过实现 K8s 调度器扩展（
+[Scheduler extender](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/scheduling/scheduler_extender.md)）来添加自定义调度规则的。
+
+TiDB Scheduler 组件部署为一个或者多个 Pod，但同时只有一个 Pod 在工作。Pod 内部有两个 Container，一个 Container 是原生的 `kube-scheduler`，另外一个 Container 是 `tidb-scheduler`，实现为一个 K8s scheduler extender。
+
+TiDB Operator 创建的所有 Pod 的 `.spec.schedulerName` 属性会被设置为
```
Comment (tennix) [inline dev/tidb-in-kubernetes/reference/components/tidb-scheduler.md]: 可以提一下，如果是测试集群，并且不要求高可用，可以将 schedulerName 改成 `default-scheduler` 使用 k8s 内置的调度器。

**Comment by yikeke** [comment]
After this PR merges, sre-bot will automatically cherry-pick this PR to the `release-3.1` and `release-3.0` branches (the needs-cherry-pick-xxx labels will trigger the behavior). FYI. @weekface @cofyc

**Comment by yikeke** [comment]
ping @tennix @cofyc Please give your LGTM if you have no further comments.

**Comment by TomShawn** [comment]
@lilin90 @ran-huang Should this document be migrated to the docs-tidb-operator repo?

**Comment by ran-huang** [comment]
> @lilin90 @ran-huang Should this document be migrated to the docs-tidb-operator repo?

@TomShawn Thanks for pointing out! I'll make a list of such documents and see how to migrate them later.

**Pull request merged**